### ISSUE TITLE
qty_invoiced on order items not set with Klarna Pay Later invoice 

### ISSUE THREAD

**Issue by mybinaryromance** [mollie/Magento#196]
When creating the invoice on Klarna Pay Later payment method, the qty_invoiced is not set for the order items. This leads to the problem that no creditmemo can be created for these orders resp. the creditmemo has an amount of 0.

Please use Mage_Sales_Model_Order_Invoice_Api for invoice creation in  app/code/community/Mollie/Mpm/Model/Client/Orders.php

**Comment by Frank-Magmodules** [comment]
Hi @mybinaryromance ,
Thank you for opening and reporting this issue and sorry for the late response to your ticket. We have created an internal task for this issue and will try to come back to you as soon as possible. Thank you for your patience!

**Comment by mybinaryromance** [comment]
I have investigated the issue a little further and would like to share my findings.
- The faulty code is taking place around Mollie_Mpm_Model_Client_Orders lines 265 (it says "Create pending invoice, as order has not been paid." in the comment).
- the call to $invoice->register() is skipped for invoices that are supposed to be captured later, that is e.g. on shipment creation. This will lead to a lot of missing data in the sales_flat_order and sales_flat_order_items database tables, like qty_invoiced, total_invoiced etc.

A quick workaround might be to call the $invoice->register() method for every invoice creation but making the capture process in Mage_Sales_Model_Order_Invoice::register() itself fail by making "if ($captureCase)" (around line 650) fail.
This might be easily achieved by rewriting Mollie_Mpm_Model_Client_Orders lines 265ff like this:


```
$invoice = $order->prepareInvoice();                            
$invoiceStatus = $this->mollieHelper->getInvoiceMomentPaidStatus($order);
                            
                            if ($invoiceStatus != \Mage_Sales_Model_Order_Invoice::STATE_PAID) { //we dont want to pay order now
								$invoice->setRequestedCaptureCase(NULL); //force no capture by causing  "if ($captureCase)" to fail, see Mage_Sales_Model_Order_Invoice::register()
								$invoice->register(); //register without capture
								$invoice->setRequestedCaptureCase(Mage_Sales_Model_Order_Invoice::CAPTURE_ONLINE); //set capture scenario for later use
							} else { //normal behaviour, capture resp pay on invoice creation
								$invoice->setRequestedCaptureCase(Mage_Sales_Model_Order_Invoice::CAPTURE_ONLINE);
								$invoice->register();
							}
                            $invoice->setTransactionId($transactionId);
                            $invoice->setState($invoiceStatus);

                            Mage::getModel('core/resource_transaction')
                                ->addObject($invoice)
                                ->addObject($invoice->getOrder())
                                ->save();
```

**Comment by Frank-Magmodules** [comment]
HI @mybinaryromance  ,

We are happy to share that we've just released the new [5.6.5](https://github.com/mollie/Magento/releases/tag/v5.6.5) version with a fix for this issue. Thank you again for the detailed reports and patience. We are closing this issue now but please feel free to reopen the issue if this still occurs.
